system-contracts/contracts/precompiles/Sekp256r1.yul

```diff
@@ -0,0 +1,120 @@
+            /// Caller should ensure the type matching before!
+            function unsafePackPrecompileParams(
+                uint32_inputOffsetInWords,
+                uint32_inputLengthInWords,
+                uint32_outputOffsetInWords,
+                uint32_outputLengthInWords,
+                uint64_perPrecompileInterpreted
+            ) -> rawParams {
+                rawParams := uint32_inputOffsetInWords
+                rawParams := or(rawParams, shl(32, uint32_inputLengthInWords))
+                rawParams := or(rawParams, shl(64, uint32_outputOffsetInWords))
+                rawParams := or(rawParams, shl(96, uint32_outputLengthInWords))
+                rawParams := or(rawParams, shl(192, uint64_perPrecompileInterpreted))
+            }
+
+            /// @dev Executes the `precompileCall` opcode.
+            function precompileCall(precompileParams, gasToBurn) -> ret {
+                // Compiler simulation for calling `precompileCall` opcode
+                ret := verbatim_2i_1o("precompile", precompileParams, gasToBurn)
+            }
+
+            ////////////////////////////////////////////////////////////////
+            //                      FALLBACK
+            ////////////////////////////////////////////////////////////////
+
+            let digest := calldataload(0)
+            let r := calldataload(32)
+            let s := calldataload(64)
+            let x := calldataload(96)
+            let y := calldataload(128)
+
+            // Validate the input by the yellow paper rules (Appendix E. Precompiled contracts)
+            let sIsInvalid := or(eq(s, 0), gt(s, sub(SECP256K1_GROUP_SIZE(), 1)))
+            let rIsInvalid := or(eq(r, 0), gt(r, sub(SECP256K1_GROUP_SIZE(), 1)))
+
+            if or(sIsInvalid, rIsInvalid) {
+                return(0, 0)
+            }
+
+            // No need for us to check whether the point is on curve as it is done in the internal precompile implementation
+
+            // Store the data in memory, so the ecrecover circuit will read it 
+            mstore(0, digest)
+            mstore(32, r)
+            mstore(64, s)
+            mstore(96, x)
+            mstore(128, y)
+
+            let precompileParams := unsafePackPrecompileParams(
+                0, // input offset in words
+                5, // input length in words (the signed digest, r, s, x, y)
+                0, // output offset in words
+                1, // output length in words (success)
+                0  // No special meaning, sekp256r1 circuit doesn't check this value
+            )
+            let gasToPay := SEKP256_VERIFY_GAS_COST()
+
+            // Check whether the call is successfully handled by the ecrecover circuit
+            let success := precompileCall(precompileParams, gasToPay)
+            let internalSuccess := mload(0)
+
+            switch and(success, internalSuccess)
+            case 0 {
+                return(0, 0)
+            }
```

Comment:
I haven't found that it should return empty result in the original EIP/RIP but it seems to be true based on other implementation I found. Are you sure it is never revert or return `0x00..00`?

Reply:
Yes, I received confirmation from the author of the EIP

Reply:
Here is the latest reference implementation: https://github.com/ulerdogan/go-ethereum/pull/1/files

It should also return `0x` if the length of the input is not correct, so I'll add it also